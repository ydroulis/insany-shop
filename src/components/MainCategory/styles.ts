'use client'

import styled from 'styled-components'

export const Container = styled.main`
    padding: 5rem 10rem 7.188rem 10rem;

    font-family: Inter, sans-serif
`