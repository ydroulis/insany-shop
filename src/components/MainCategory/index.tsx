import React from 'react';

import * as S from './styles';
import { Products } from '@/types/Products';
import Filters from '../Filters';
import ProductList from '../ProductList';
import { useProductsStore } from '../../providers/productsStoreProvider';
import { useCategoriesStore } from '../../providers/categoriesStoreProvider';

interface MainCategoryProps {
    pageId: string;
}

const MainCategory: React.FC<MainCategoryProps> = ({ pageId }) => {
    const { products } = useProductsStore((state => state));
    const { categories } = useCategoriesStore((state => state));

    const productList: Products = products.map(product => ({
        ...product,
        category: categories.find(category => category.id === product.category)?.name
    }))
    return (
        <S.Container
            as="main"
            role="main"
            aria-label={`CatÃ¡logo de produtos ${pageId}`}
        >
            <Filters pageId={pageId} />
            <ProductList products={productList} pageId={pageId} />
        </S.Container>
    );
}

export default MainCategory;