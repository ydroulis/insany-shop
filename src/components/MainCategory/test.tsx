import React from "react";
import { render, screen } from "@testing-library/react";
import MainCategory from "./";


describe("<MainCategory />", () => {
    beforeEach(() => {
        render(<MainCategory pageId="1" />);
    });

    it("should render the main landmark with correct accessibility attributes", () => {
        const main = screen.getByRole("main", { name: /catÃ¡logo de produtos/i });
        expect(main).toBeInTheDocument();
        expect(main.tagName.toLowerCase()).toBe("main");
    });
    it("should match snapshot", () => {
        const { container } = render(<MainCategory pageId="1" />);
        expect(container).toMatchSnapshot();
    });
});
