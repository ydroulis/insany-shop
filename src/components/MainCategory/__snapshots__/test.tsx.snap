// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<MainCategory /> should match snapshot 1`] = `
<div>
  <main
    aria-label="Catálogo de produtos 1"
    class="sc-bRKDuR dRZDNv"
    role="main"
  >
    <section
      aria-describedby="product-list-description"
      aria-labelledby="product-list-title"
      class="sc-hjsuWn bOYdgP"
    >
      <h2
        class="sc-jJLAfE kxToVy"
        id="product-list-title"
      />
      <p
        class="sc-hwkwBN cgVEZH"
        id="product-list-description"
      />
      <p
        aria-live="polite"
        role="status"
      >
        Nenhum produto disponível no momento.
      </p>
    </section>
  </main>
</div>
`;
